File: app/src/main/java/io/github/wulkanowy/timetable/InputFragment.java
package io.github.wulkanowy.timetable;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;
import android.widget.Button;

import java.util.LinkedHashMap;

public class InputFragment extends Fragment {

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_input, container, false);
        final AutoCompleteTextView schoolView = view.findViewById(R.id.school);

        // Get the string array
        String[] timetables = view.getResources().getStringArray(R.array.timetables_schools);
        // Create the adapter and set it to the AutoCompleteTextView
        ArrayAdapter<String> adapter = new ArrayAdapter<>
                (getActivity(), android.R.layout.simple_list_item_1, timetables);
        schoolView.setAdapter(adapter);

        Button button = view.findViewById(R.id.action_save);
        button.setOnClickListener(new OnClickListener() {
            public void onClick(View v) {
                String timetableUrl = schoolView.getText().toString();

                String[] keys = getResources().getStringArray(R.array.timetables_schools);
                String[] values = getResources().getStringArray(R.array.timetables_urls);
                LinkedHashMap<String, String> map = new LinkedHashMap<>();

                for (int i = 0; i < Math.min(keys.length, values.length); ++i) {
                    map.put(keys[i], values[i]);
                }

                if (map.containsKey(timetableUrl)) {
                    timetableUrl = map.get(timetableUrl);
                }

                Snackbar.make(getActivity().findViewById(R.id.container),
                        timetableUrl, Snackbar.LENGTH_SHORT).show();
            }
        });

        return view;
    }
}

File: app/src/main/kotlin/io/github/wulkanowy/timetable/InputFragment.kt
package io.github.wulkanowy.timetable

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.fragment.app.Fragment
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.fragment_input.*
import java.util.*

class InputFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_input, container, false)

        // Get the string array
        val timetables = view.resources.getStringArray(R.array.timetables_schools)
        // Create the adapter and set it to the AutoCompleteTextView
        val adapter = ArrayAdapter(activity!!, android.R.layout.simple_list_item_1, timetables)
        school.setAdapter(adapter)

        actionSave.setOnClickListener {
            var timetableUrl = school.text.toString()

            val keys = resources.getStringArray(R.array.timetables_schools)
            val values = resources.getStringArray(R.array.timetables_urls)
            val map = LinkedHashMap<String, String>()

            for (i in 0 until Math.min(keys.size, values.size)) {
                map[keys[i]] = values[i]
            }

            if (map.containsKey(timetableUrl)) {
                timetableUrl = map[timetableUrl] as String
            }

            Snackbar.make(activityContainer, timetableUrl, Snackbar.LENGTH_SHORT).show()
        }

        return view
    }
}
